src/prefect/deployments.py

```diff
@@ -375,7 +377,7 @@ def _yaml_dict(self) -> dict:
         description="The work queue for the deployment.",
         yaml_comment="The work queue that will handle this deployment's runs",
     )
-
+    worker_pool_queue_id: Optional[UUID] = Field(default=None)
```

Comment:
Might be worth adding an `ExperimentalField` utility that uses our warning/error patterns.

Reply:
I submitted a method for marking experimental fields in https://github.com/PrefectHQ/prefect/pull/8066. Will mark this field as experimental once that PR is merged.